bgzf.c

```diff
@@ -805,7 +805,8 @@ int bgzf_read_block(BGZF *fp)
                 if (fp->uncompressed_block == NULL) return -1;
                 fp->compressed_block = (char *)fp->uncompressed_block + BGZF_MAX_BLOCK_SIZE;
             } // else it's already allocated with malloc, maybe even in-use.
-            mt_destroy(fp->mt);
+            if (mt_destroy(fp->mt) < 0)
+                fp->errcode = BGZF_ERR_IO;
```

Comment:
Hmm I'm wondering if this has the same missing mutex as the bgzf_mt_writer code's previous attempt at setting errcode.  I didn't validate this using libdeflate.  However it does seem to not produce whinges from -fsanitize=thread (unlike the original bgzf_mt_writer func).  So I'm assuming it's not in a situation that can trigger data races.